src/scene/Entity.cpp

```diff
@@ -77,3 +84,60 @@ std::optional<Mat3x4f> Entity::getPreviousFrameLocalToWorldTransform() const
 
 	return formerTransformInfo.matrix;
 }
+
+void Entity::applyExternalAnimation(const Vec3f* vertices, std::size_t vertexCount)
+{
+	if (!std::holds_alternative<ExternalAnimator>(animator)) {
+		animator = ExternalAnimator(mesh->dVertices);
+	}
+	if (auto* externalAnimator = std::get_if<ExternalAnimator>(&animator)) {
```

Comment:
The only alternative is a shared interface between animators, but since we cannot define a single update function that would be overridden (because update functions for each animator have different arguments) the variant is better here.